language-plutus-core/src/Language/PlutusCore/MkPlc.hs

```diff
@@ -52,6 +52,8 @@ class TermLike term tyname name | term -> tyname, term -> name where
     unwrap   :: a -> term a -> term a
     iWrap    :: a -> Type tyname a -> Type tyname a -> term a -> term a
     error    :: a -> Type tyname a -> term a
+    termLet  :: a -> TermDef term tyname name a -> term a -> term a
+    typeLet  :: a -> TypeDef tyname a -> term a -> term a
```

Comment:
Maybe then make `termLet = mkTermLet` the default, since `mkTermLet` is generic enough? And same for `typeLet`. I would even inline the definition of `mkTermLet` here.

Reply:
Yeeees although it doesn't gain us anything since we have precisely two implementations, one of which is different. I also want to have the top-level `mkTermLet` since we need to use it during compilation. Maybe we should rename it.